ir/marshal.go

```diff
@@ -0,0 +1,79 @@
+package ir
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+)
+
+type MarshalType int
+
+// List of types
+const (
+	StringType = iota
+	BlobType
+	PeerType
+	IntType
+	FloatType
+	BoolType
+)
+
+// UnmarshalType list of supported types
+func UnmarshalType(tp MarshalType, b []byte) (Node, error) {
```

Comment:
Document what is b. Is this the encoding of the type/value-wrapper, or is it the encoding just of the value? It looks like it is just the encoding of the value (no wrapper).